spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -302,6 +302,7 @@ protected void doInitialize() throws Exception {
 		if (this.taskScheduler == null) {
 			ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
 			threadPoolTaskScheduler.setThreadNamePrefix(getBeanName() + "-consumerMonitor-");
+			threadPoolTaskScheduler.setDaemon(true);
```

Comment:
Right, something like `boolean taskSchedulerExplicitlySet`. We use such an approach in other places.

Otherwise we should revise the `shutdown` logic, because of just:
```
public void destroy() {
	shutdown();
}
```
In the `AbstractMessageListenerContainer`. 
I see that we call `shutdown();` in the `actualStart()`. Might be dangerous at some point.

OTOT re. `daemon`. I would like to understand how it finishes to schedule then, if we don't call any shutdown at all...